diff --git a/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsModule.kt b/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsModule.kt
index 981546b..dd9bad0 100644
--- a/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsModule.kt
+++ b/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsModule.kt
@@ -982,7 +982,7 @@ class ReactNativeFsModule internal constructor(context: ReactApplicationContext)
             promise.reject(ex.code, ex.message)
             return
         }
-        promise.reject(null, ex!!.message)
+        promise.reject("ERROR", ex!!.message)
     }
 
     private fun rejectFileNotFound(promise: Promise, filepath: String?) {
