diff --git a/node_modules/@internxt/mobile-sdk/android/build.gradle b/node_modules/@internxt/mobile-sdk/android/build.gradle
index a5869fd..9e27916 100644
--- a/node_modules/@internxt/mobile-sdk/android/build.gradle
+++ b/node_modules/@internxt/mobile-sdk/android/build.gradle
@@ -16,7 +16,7 @@ buildscript {
   }
 }
 plugins {
-  id "com.google.devtools.ksp" version "1.8.20-1.0.10"
+  id "com.google.devtools.ksp" version "1.9.25-1.0.20"
 }
 
 def isNewArchitectureEnabled() {
