diff --git a/node_modules/react-native-image-picker/ios/ImagePickerUtils.mm b/node_modules/react-native-image-picker/ios/ImagePickerUtils.mm
index 443500d..6f8d0ba 100644
--- a/node_modules/react-native-image-picker/ios/ImagePickerUtils.mm
+++ b/node_modules/react-native-image-picker/ios/ImagePickerUtils.mm
@@ -97,6 +97,8 @@
     const uint8_t firstByteJpg = 0xFF;
     const uint8_t firstBytePng = 0x89;
     const uint8_t firstByteGif = 0x47;
+    const uint8_t firstByteWebp = 0x52;
+    const uint8_t firstByteHeic = 0x00;
     
     uint8_t firstByte;
     [imageData getBytes:&firstByte length:1];
@@ -107,6 +109,10 @@
         return @"png";
       case firstByteGif:
         return @"gif";
+      case firstByteWebp:
+        return @"webp";
+      case firstByteHeic:
+        return @"heic";
       default:
         return @"jpg";
     }
